### ISSUE TITLE
Crash when loading model from ruby (munmap_chunk(): invalid pointer)

### ISSUE THREAD

**Issue by oblivione81** [abrandoned/ffi-fasttext#4]
I'm using ffi-fasttext 0.1.1 with ruby 2.7, installed through bundler, on a machine with g++ 7.3.1.
fasttext is correctly built as a .so object as a native extension. I'm not setting FASTTEXT_USE_SYSTEM_LIB or FASTTEXT_LIB_PATH.

I can find libfasttext.so it in the gem under `bundle/ruby/2.7.0/gems/ffi-fasttext-0.1.1/vendor/fasttext/libfasttext.so`, and I confirm that's the one that gets loaded upon initialization, but, whenever I try to load a valid model with `FFI::Fasttext::Predictor.new(path)`, from a ruby console, I get this failure (which causes the whole console to crash):
```
munmap_chunk(): invalid pointer
Aborted
```

The model should be fine, since I can successfully use the fasttext executable, built in the same folder as the .so library, with the same model, and it correctly performs a  prediction.

Also, more interestingly, I have another machine, with a pretty identical configuration, the biggest difference being the version of g++ being 4.8.5 rather than 7.3.1. 
On that machine fasttext is working well on the same model and, if I replace libfastext.so on the faulty machine with the one from the working machine, everything works fine.   

So basically, from what I understand, it looks like there might be a problem in the connection between ruby and the native lib when using g++ 7.3.1
